src/app/core/auth/auth.service.test.ts

```diff
@@ -0,0 +1,55 @@
+import { AuthService } from './auth.service';
+
+describe('AuthService', function () {
+    let shHttpMock;
+    let profileMock;
+    let service;
+
+    beforeEach(() => {
+        shHttpMock = jasmine.createSpyObj('shHttpMock', ['post']);
+        profileMock = jasmine.createSpyObj('profileMock', ['setUserData', 'removeUserData']);
+        service = new AuthService(shHttpMock, profileMock);
+    });
+
+    describe('#login', () => {
+        it('should login and set user data', () => {
+            // Data
+            const authApiUrl = '/auth/login';
+            const credentials = {
+                email: 'valid@valid.com',
+                password: 'valid-pass'
+            };
+            const authLoginPostData = {
+                user: {},
+                token: 'test-token'
+            };
+
+            // Spies
+            const httpPostObservable = {
+                map(next) {
+                    next(authLoginPostData);
+                    return this;
+                }
+            };
+            shHttpMock.post.and.returnValue(httpPostObservable);
+
+            // Run
+            const result = service.login(credentials);
+
+            // Expect
+            expect(result).toBe(httpPostObservable);
```

Comment:
Different assertion related to one case can be grouped with describe section. But it is better when one assertion can fail only one test